src/Microsoft.Health.Dicom.Api/Features/Context/DicomRequestContextMiddleware.cs

```diff
@@ -51,8 +57,24 @@ public async Task Invoke(HttpContext context, IDicomRequestContextAccessor dicom
 
             dicomRequestContextAccessor.RequestContext = dicomRequestContext;
 
-            // Call the next delegate/middleware in the pipeline
-            await _next(context);
+            ByteCountingStream byteCountingStream = _responseLogStreamFactory.CreateByteCountingResponseLogStream(context.Response.Body);
+            context.Response.Body = byteCountingStream;
+
+            try
+            {
+                // Call the next delegate/middleware in the pipeline
+                await _next(context);
+            }
+            finally
+            {
+                long responseBodySize = byteCountingStream.WrittenByteCount;
+                long responseHeaderSize = HeaderUtility.GetTotalHeaderLength(context.Response.Headers);
+                long totalResponseSize = responseBodySize + responseHeaderSize;
+                context.Items["ResponseSize"] = totalResponseSize;
```

Comment:
Should we use a constant for the item keys?

Reply:
Removed